### DISCUSSION TITLE
IOTC_ER_TIMEOUT

### DISCUSSION THREAD

**Discussion by christianhjohnson** [mrlt8/docker-wyze-bridge#1105] [General]
I am trying to bridge a camera located on a different network. I am getting an IOTC_ER_TIMEOUT

2024/01/26 16:32:56 | stdout | [WyzeBridge] 172.17.0.1 - - [27/Jan/2024 00:32:56] "[36mGET /img/kathryn-cam-pan-3.jpg HTTP/1.1[0m" 304 -
-- | -- | --
2024/01/26 16:32:56 | stdout | [WyzeBridge] 172.17.0.1 - - [27/Jan/2024 00:32:56] "[36mGET /img/kathryn-cam-pan-3.jpg HTTP/1.1[0m" 304 -
2024/01/26 16:32:26 | stdout | [WyzeBridge] 172.17.0.1 - - [27/Jan/2024 00:32:26] "[36mGET /img/kathryn-cam-pan-3.jpg HTTP/1.1[0m" 304 -
2024/01/26 16:32:26 | stdout | [WyzeBridge] 172.17.0.1 - - [27/Jan/2024 00:32:26] "[36mGET /img/kathryn-cam-pan-3.jpg HTTP/1.1[0m" 304 -
2024/01/26 16:32:16 | stdout | [kathryn-cam-pan-3] [-13] IOTC_ER_TIMEOUT
2024/01/26 16:32:11 | stdout | [WyzeBridge] 172.17.0.1 - - [27/Jan/2024 00:32:11] "[36mGET /img/kathryn-cam-pan-3.jpg HTTP/1.1[0m" 304 -
2024/01/26 16:32:11 | stdout | [WyzeBridge] 172.17.0.1 - - [27/Jan/2024 00:32:11] "[36mGET /img/kathryn-cam-pan-3.jpg HTTP/1.1[0m" 304 -
2024/01/26 16:32:11 | stdout | [WyzeBridge] ☁️ Pulling "kathryn-cam-pan-3" thumbnail to /img/kathryn-cam-pan-3.jpg
2024/01/26 16:32:11 | stdout | [WyzeBridge] [kathryn-cam-pan-3] Snapshot timed out
2024/01/26 16:31:56 | stdout | [WyzeBridge] 172.17.0.1 - - [27/Jan/2024 00:31:56] "GET /api/sse_status HTTP/1.1" 200 -
2024/01/26 16:31:56 | stdout | [WyzeBridge] 172.17.0.1 - - [27/Jan/2024 00:31:56] "GET /api/sse_status HTTP/1.1" 200 -
2024/01/26 16:31:56 | stdout | [WyzeBridge] 🎉 Connecting to WyzeCam Pan V3 - Kathryn Cam Pan 3 on 10.242.150.62
2024/01/26 16:31:56 | stdout | [WyzeBridge] 172.17.0.1 - - [27/Jan/2024 00:31:56] "GET /static/loading.svg HTTP/1.1" 200 -
2024/01/26 16:31:56 | stdout | [WyzeBridge] 172.17.0.1 - - [27/Jan/2024 00:31:56] "[36mGET /static/bulma.css HTTP/1.1[0m" 304 -
2024/01/26 16:31:56 | stdout | [WyzeBridge] 172.17.0.1 - - [27/Jan/2024 00:31:56] "[36mGET /static/site.css HTTP/1.1[0m" 304 -
2024/01/26 16:31:56 | stdout | [WyzeBridge] 172.17.0.1 - - [27/Jan/2024 00:31:56] "[36mGET /static/bulma-toast.js HTTP/1.1[0m" 304 -
2024/01/26 16:31:56 | stdout | [WyzeBridge] 172.17.0.1 - - [27/Jan/2024 00:31:56] "[36mGET /static/site.js HTTP/1.1[0m" 304 -
2024/01/26 16:31:56 | stdout | [WyzeBridge] 172.17.0.1 - - [27/Jan/2024 00:31:56] "GET / HTTP/1.1" 200 -
2024/01/26 16:31:56 | stdout | [WyzeBridge] 🎬 1 stream enabled
2024/01/26 16:31:56 | stdout | [WyzeBridge] starting MediaMTX 1.1.1
2024/01/26 16:31:56 | stdout | [WyzeBridge] [+] Adding Kathryn Cam Pan 3 [HL_PAN3]
2024/01/26 16:31:56 | stdout | [WyzeBridge] 💾 Saving 'cameras' to local cache...
2024/01/26 16:31:56 | stdout | [WyzeBridge] [API] Fetched [1] cameras
2024/01/26 16:31:55 | stdout | [WyzeBridge] ☁️ Fetching 'cameras' from the Wyze API...
2024/01/26 16:31:55 | stdout | [WyzeBridge] 🔍 Could not find local cache for 'cameras'
2024/01/26 16:31:55 | stdout | [WyzeBridge] 💾 Saving 'user' to local cache...
2024/01/26 16:31:55 | stdout | [WyzeBridge] ☁️ Fetching 'user' from the Wyze API...
2024/01/26 16:31:55 | stdout | [WyzeBridge] 🔍 Could not find local cache for 'user'
2024/01/26 16:31:55 | stdout | [WyzeBridge] 💾 Saving 'auth' to local cache...
2024/01/26 16:31:55 | stdout | [WyzeBridge] [33mPress CTRL+C to quit[0m
2024/01/26 16:31:55 | stdout | * Running on http://172.17.0.2:5000
2024/01/26 16:31:55 | stdout | * Running on http://127.0.0.1:5000
2024/01/26 16:31:55 | stdout | * Running on all addresses (0.0.0.0)
2024/01/26 16:31:55 | stdout | [WyzeBridge] [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
2024/01/26 16:31:55 | stdout | * Debug mode: off
2024/01/26 16:31:55 | stdout | * Serving Flask app 'frontend'
2024/01/26 16:31:55 | stdout | [WyzeBridge] ☁️ Fetching 'auth' from the Wyze API...
2024/01/26 16:31:55 | stdout | [WyzeBridge] 🔍 Could not find local cache for 'auth'
2024/01/26 16:31:55 | stdout |  
2024/01/26 16:31:55 | stdout | 🚀 DOCKER-WYZE-BRIDGE v2.6.0 X86_64 2.7.0 BUILD [2024-01-11t04:20:13.342z] bbacb0f

![image](https://github.com/mrlt8/docker-wyze-bridge/assets/89424695/01a26109-acd4-46a7-a7d0-131db9176f53)

I get this when accessing the bridge.

I am able to access my local cameras via the bridge.

I had the bridge running a while and was recording 24/7 on my synology, but it seems like I can't re-connect.

Any ideas?

**Comment by quochung9999** [comment]
i have the same issues when updating to the latest firmware. I have to flash them back to the original wyze firmware that support the rtsp streaming (version 4.61.0.3). that fixed them all.

**Comment by tomcom05** [comment]
I can confirm that rolling back the FW to [4.50.4.7252] fixes this issue for my camera.

**Comment by live2pipe** [comment]
I was having this issue too.  After many, many hours of log reviews and experimenting with different configurations running in both a standalone container, and as a HA Add On, I came to realize that when I change the docker network used by the container to a "macvlan" type, the application runs GREAT!!!  Best it has ever run for me in the last year.  I'm not sure why this is the case but wanted to pass along this info in case it helps others.

**Reply by IrvensAntoine** [reply]
Where did you change the container to use macvlan type?

**Reply by letrain02** [reply]
> I was having this issue too.  After many, many hours of log reviews and experimenting with different configurations running in both a standalone container, and as a HA Add On, I came to realize that when I change the docker network used by the container to a "macvlan" type, the application runs GREAT!!!  Best it has ever run for me in the last year.  I'm not sure why this is the case but wanted to pass along this info in case it helps others.

So for me. Host mode they all work, except no webrtc. Switched to bridge. Nothing works. Changed to macvlan with bridge and 2 cameras work and webrtc... So not best results for me. Host mode is the only way to get all cameras to connect, macvlan w/bridge mode is the only way to use webrtc...

**Reply by IrvensAntoine** [reply]
I was finally able to get it to work with the local host bridge add-on. My webrtc cameras weren't working initially as well. I noticed it was trying to use port 8554 which was conflicting with the bridge initially; however, removing and re-adding webrtc (after installing the local host bridge) appears to run fine at the same time with the bridge using port 8554. All my webrtc cams were using 8553 prior to this issue popping up. I can't remember for the life of me how I initially had it set to 8553. The reinstall forces 8554, but again still appears to run along side the local host bridge, that's also using 8554.

**Comment by live2pipe** [comment]
On a standalone docker host (not associated with Home Assistant) I created a macvlan using:

`docker network create --driver macvlan --subnet 192.168.1.0/24 --gateway 192.168.1.1 --ip-range=192.168.1.25/32 -o parent=enp1s0 wyze_bridge_macvlan_network`

in my case I wanted the macvlan to bind to a specific address on my lan (192.168.1.25) hence the `--ip-range=192.168.1.25/32` option.  Without specifying `--ip-range=`, macvlan will bind to the first ip address in the subnet.  In my case that was 192.168.1.2 which was in  use by another device.

then i referenced the network in the network_mode option of my docker-compose.yml :

```
services:
    wyze-bridge:
        container_name: wyze-bridge
        restart: unless-stopped
        image: mrlt8/wyze-bridge:latest
        network_mode: wyze_bridge_macvlan_network

        ports:
            - 1936:1935 # RTMP
            - 8556:8554 # RTSP
            - 8888:8888 # HLS
            - 8889:8889 #WebRTC
            - 8189:8189/udp # WebRTC/ICE
            - 5010:5000 # WEB-UI
```

I ran `docker-compose up` (no -d option) so I could monitor the log messages.  Everything worked perfectly.  As soon as I removed the `network_mode` refererence to my macvlan network and restarted the container, I'd get IOTC Timeouts repeatedly and the cameras streams would never come up.

I tried dozens docker-compose scenarios but this was the only on way i could get it working.

I haven't been able to figure out how to do something similar on Home Assistant when docker-wyze-bridge is running as an add_on.  I assume there's a docker compose file somewhere burried inside Home Assistant but I haven't been able to figure out where it lives.

**Reply by Anc0dia** [reply]
Thanks for posting this.   I tried setting up a macvlan using Portainer and it didn't work, but using your command line did work, and wyzebridge is now back online.

* edit:  hmm, maybe it won't work - I can't communicate to the macvlan from other containers on the same host - i.e if I have frigate @ 192.168.200.15 and the wyzebridge on a macvlan using 192.168.200.20 both using the same physical nic they can't ping each other.

*edit 2: so I would need to add all the containers to the macvlan network

create the macvlan

docker network create --driver macvlan --subnet 192.168.200.0/24 --gateway 192.168.200.254 --ip-range=192.168.200.0/24 -o parent=enp1s0 wyzebridge_macvlan

then modify each of the containers and add:

```
networks:
  whatever-local:
    external: true
    name: wyzebridge-macvlan

services:
        networks:
          whatever-local:
            ipv4_address: 192.168.200.20 # ip of server

```

**Reply by Roachforlife** [reply]
> On a standalone docker host (not associated with Home Assistant) I created a macvlan using:
> 
> `docker network create --driver macvlan --subnet 192.168.1.0/24 --gateway 192.168.1.1 --ip-range=192.168.1.25/32 -o parent=enp1s0 wyze_bridge_macvlan_network`
> 
> in my case I wanted the macvlan to bind to a specific address on my lan (192.168.1.25) hence the `--ip-range=192.168.1.25/32` option. Without specifying `--ip-range=`, macvlan will bind to the first ip address in the subnet. In my case that was 192.168.1.2 which was in use by another device.
> 
> then i referenced the network in the network_mode option of my docker-compose.yml :
> 
> ```
> services:
>     wyze-bridge:
>         container_name: wyze-bridge
>         restart: unless-stopped
>         image: mrlt8/wyze-bridge:latest
>         network_mode: wyze_bridge_macvlan_network
> 
>         ports:
>             - 1936:1935 # RTMP
>             - 8556:8554 # RTSP
>             - 8888:8888 # HLS
>             - 8889:8889 #WebRTC
>             - 8189:8189/udp # WebRTC/ICE
>             - 5010:5000 # WEB-UI
> ```
> 
> I ran `docker-compose up` (no -d option) so I could monitor the log messages. Everything worked perfectly. As soon as I removed the `network_mode` refererence to my macvlan network and restarted the container, I'd get IOTC Timeouts repeatedly and the cameras streams would never come up.
> 
> I tried dozens docker-compose scenarios but this was the only on way i could get it working.
> 
> I haven't been able to figure out how to do something similar on Home Assistant when docker-wyze-bridge is running as an add_on. I assume there's a docker compose file somewhere burried inside Home Assistant but I haven't been able to figure out where it lives.

Just to add that I followed this and it worked. I didnt realize that I would have to lose an ip in the process (one for my lxc container and now a new one for the wyze bridge docker container) but its all good, and now my V2 works again. Thanks!  My eight V3s do not have this issue only the V2

**Reply by cheme75** [reply]
I assume this is on a Linux OS?

**Reply by cheme75** [reply]
> On a standalone docker host (not associated with Home Assistant) I created a macvlan using:
> 
> 
> 
> `docker network create --driver macvlan --subnet 192.168.1.0/24 --gateway 192.168.1.1 --ip-range=192.168.1.25/32 -o parent=enp1s0 wyze_bridge_macvlan_network`
> 
> 
> 
> in my case I wanted the macvlan to bind to a specific address on my lan (192.168.1.25) hence the `--ip-range=192.168.1.25/32` option.  Without specifying `--ip-range=`, macvlan will bind to the first ip address in the subnet.  In my case that was 192.168.1.2 which was in  use by another device.
> 
> 
> 
> then i referenced the network in the network_mode option of my docker-compose.yml :
> 
> 
> 
> ```
> 
> services:
> 
>     wyze-bridge:
> 
>         container_name: wyze-bridge
> 
>         restart: unless-stopped
> 
>         image: mrlt8/wyze-bridge:latest
> 
>         network_mode: wyze_bridge_macvlan_network
> 
> 
> 
>         ports:
> 
>             - 1936:1935 # RTMP
> 
>             - 8556:8554 # RTSP
> 
>             - 8888:8888 # HLS
> 
>             - 8889:8889 #WebRTC
> 
>             - 8189:8189/udp # WebRTC/ICE
> 
>             - 5010:5000 # WEB-UI
> 
> ```
> 
> 
> 
> I ran `docker-compose up` (no -d option) so I could monitor the log messages.  Everything worked perfectly.  As soon as I removed the `network_mode` refererence to my macvlan network and restarted the container, I'd get IOTC Timeouts repeatedly and the cameras streams would never come up.
> 
> 
> 
> I tried dozens docker-compose scenarios but this was the only on way i could get it working.
> 
> 
> 
> I haven't been able to figure out how to do something similar on Home Assistant when docker-wyze-bridge is running as an add_on.  I assume there's a docker compose file somewhere burried inside Home Assistant but I haven't been able to figure out where it lives.

Is the .25 address the server or an open/available address?

Assuming this is on Linux, not Win?

**Comment by jeeftor** [comment]
I'm having this error today as well ... I can see snapshots of the camera but can't stream

**Comment by littlekools** [comment]
I beleave I am having the same problem with RTSP (runing on HA addon) as this thread and its been a few weeks. the strange thing is that I have 4 cams running (v2, v3, v3, and v1pan) and the v1pan still works with RTSP

[WyzeBridge] 🎉 Connecting to WyzeCam V3 - front right on 192.168.50.189
[WyzeBridge] 🎉 Connecting to WyzeCam V3 - Front Left on 192.168.50.4
[WyzeBridge] 🎉 Connecting to WyzeCam V2 - patio on 192.168.50.39
[WyzeBridge] ⏰ Timed out connecting to front right.
[front-left] [-13] IOTC_ER_TIMEOUT
[front-right] [-13] IOTC_ER_TIMEOUT
[patio] [-13] IOTC_ER_TIMEOUT

**Comment by pmatioli** [comment]
Hello,
  
 I'm running wyze-bridge in RBP5 and having same issue about 1 to 2 weeks now. I've created macvlan network for my container as below. I'm still running the same issue. 

 docker network create --driver macvlan --subnet 192.168.4.0/24 --gateway 192.168.4.1  -o parent=eth0 wyze_bridge_macvlan_network

  wyze-bridge:
    image: mrlt8/wyze-bridge:latest
    container_name: wyze-bridge
    restart: unless-stopped
    network_mode: wyze_bridge_macvlan_network
    environment:
      - RTSP_ENABLED=true
      - WEBRTC_ENABLED=true
      - DEBUG_LEVEL=info
      - AUTH_ENABLED=false
      - WB_AUTH=false
    ports:
      - "1935:1935"
      - "8554:8554"
      - "8888:8888"
      - "8889:8889"
      - "8189:8189/udp"
      - "5000:5000"
      
 wyze-bridge  | [WyzeBridge] ☁️ Pulling "living-room-cam" thumbnail to /img/living-room-cam.jpg
wyze-bridge  | [WyzeBridge] 192.168.4.167 - - [02/Mar/2025 04:32:44] "GET /snapshot/living-room-cam.jpg?1740889943788 HTTP/1.1" 200 -
wyze-bridge  | [WyzeBridge] 🎉 Connecting to WyzeCam Pan V3 - Living Room Cam on 192.168.4.31
wyze-bridge  | [driveway-cam] [-13] IOTC_ER_TIMEOUT
wyze-bridge  | [back-yard-cam] [-13] IOTC_ER_TIMEOUT
wyze-bridge  | [WyzeBridge] 192.168.4.167 - - [02/Mar/2025 04:32:54] "GET /snapshot/garage-cam.jpg?1740889943788 HTTP/1.1" 304 -
wyze-bridge  | [WyzeBridge] 192.168.4.167 - - [02/Mar/2025 04:32:54] "GET /snapshot/sparecam.jpg?1740889943788 HTTP/1.1" 307 -
wyze-bridge  | [WyzeBridge] 192.168.4.167 - - [02/Mar/2025 04:32:54] "GET /static/notavailable.svg HTTP/1.1" 304 -
wyze-bridge  | [WyzeBridge] [living-room-cam] Snapshot timed out
wyze-bridge  | [WyzeBridge] 192.168.4.167 - - [02/Mar/2025 04:32:59] "GET /snapshot/living-room-cam.jpg?1740889943788 HTTP/1.1" 304 -
wyze-bridge  | [living-room-cam] [-13] IOTC_ER_TIMEOUT

**Comment by jeeftor** [comment]
I think we are all suffering currently

**Reply by johnnyv5** [reply]
For those that have Ubiquiti WiFi, I had to turn of Multicast and Broadcast Control for the SSID.  After turning this off and making sure I had the Docker Wyze Bridge configured with Host, the cameras started working and continue to work.

**Comment by samjage** [comment]
Everyone, I can confirm the following works now:

Note:
- I have the RTSP URL's setup in a scrypted container under the Scrypted RTSP CAMERA PLUGIN
- As many noted above, Host Network is required, not sure how mine was working on my "wyze-bridge_default" network container before.

Portainer:
- Duplicate Edit your Wyze-Bridge Container 
![image](https://github.com/user-attachments/assets/0a26e7a6-db1a-4c0b-976c-af314ca551a9)
- Set Network as Host from the drop down
![image](https://github.com/user-attachments/assets/7a86bffa-9b0f-47a4-911e-6bb7b7707c79)
- Enable NET_BIND_SERVICE under Capabilities (usually set by default)
![image](https://github.com/user-attachments/assets/7bfd5563-22b1-4e60-8503-e846b17435a3)
- Deploy the Container
![image](https://github.com/user-attachments/assets/957105aa-961e-4ef7-8919-bf64496f352e)

Everything now connects on Wyze Bridge x.x.x.x:5000
![image](https://github.com/user-attachments/assets/6287f80b-a6f4-41b8-9959-7a76c4ed5679)

**Reply by jeeftor** [reply]
- Does just setting `privileged` mode also work?

**Reply by samjage** [reply]
I didn't try that, but I would imagine it might.

**Comment by Abcmsaj** [comment]
Just to chime in that I'm in a similar boat. Noticed that my Scrypted streams weren't deploying to Apple Home today so went to investigate and the problem stems from Wyze Docker... I'm running the HA add-on (used to be Docker but I found the HA one more manageable as I was also using Scrypted inside HA)

![Screenshot 2025-03-08 at 11 00 22](https://github.com/user-attachments/assets/abd7d61a-812c-4aa4-9dc1-e9214c561b66)

All of my cameras are timing out. I seem to get recent-ish snapshots for 3 out of 4 of them, but the last snapshot I got for the back garden camera was 14 days ago. None of them broadcast video via rtsp. Nothing I do in the config seems to change anything, and I have tried to re-enable the Docker container I used to have with the latest compose example on this repo and even with `network_mode: host`, I get nothing :/

**Reply by Abcmsaj** [reply]
![Screenshot 2025-03-13 at 09 26 14](https://github.com/user-attachments/assets/3ae78118-6751-4b26-8f87-2440ea046928)

**Reply by maxfield-allison** [reply]
check out this thread 
https://github.com/mrlt8/docker-wyze-bridge/discussions/1471#discussion-8189896

**Comment by nranderson** [comment]
I also had the same issue as above. Fixed it by adding `--net=host` to my `docker run` command.

**Reply by raceconditions** [reply]
I can confirm this worked for me.  I changed my `docker_compose.yml` to add `network_mode: "host"` and I'm streaming again no issues.

```
version: '2.4'
services:
    wyze-bridge:
        container_name: wyze-bridge
        restart: unless-stopped
        image: mrlt8/wyze-bridge:latest
        network_mode: "host"

...
```

**Reply by letrain02** [reply]
I can confirm this worked for me. But bridging and giving the container its own ip address also made webrtc work. It's flawless again.

**Reply by Abcmsaj** [reply]
![Screenshot 2025-03-13 at 09 26 14](https://github.com/user-attachments/assets/f1b07b81-2880-48a2-ad65-92959f8d696f)

Oddly, doesn't work for me!

**Reply by letrain02** [reply]
> ![Screenshot 2025-03-13 at 09 26 14](https://github.com/user-attachments/assets/f1b07b81-2880-48a2-ad65-92959f8d696f)
> 
> Oddly, doesn't work for me!

Try assigning the container an IP in the subnet or vlan as your cams. Host worked ok for me, but giving it an IP fixed all my issues. 

For example 

--net='br0' --ip='192.168.88.9'  instead of net=`host'

Your usage may vary. And you'd need a bridged interface to reference.

**Reply by bougau** [reply]
How can I try this fix when using docker-wyze-bridge as an add-on?

**Reply by iveykun** [reply]
> > ![Screenshot 2025-03-13 at 09 26 14](https://private-user-images.githubusercontent.com/5143023/422271628-f1b07b81-2880-48a2-ad65-92959f8d696f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDIxNzYxMDIsIm5iZiI6MTc0MjE3NTgwMiwicGF0aCI6Ii81MTQzMDIzLzQyMjI3MTYyOC1mMWIwN2I4MS0yODgwLTQ4YTItYWQ2NS05Mjk1OWY4ZDY5NmYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDMxNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAzMTdUMDE0MzIyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NDczOGM2ZjdkMGRlMDAzN2I5NDcyNTExN2M0M2UzMmE1NDdlYjE5NDhmZDA3ZmZhYWQ2YTZhNzFlNTY0ZmUzMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.9baG1iOE64Vxq-jjHTWWR2Fp5Pdvv4irzLQq7Drz4ps)
> > Oddly, doesn't work for me!
> 
> Try assigning the container an IP in the subnet or vlan as your cams. Host worked ok for me, but giving it an IP fixed all my issues.
> 
> For example
> 
> --net='br0' --ip='192.168.88.9' instead of net=`host'
> 
> Your usage may vary. And you'd need a bridged interface to reference.

What's the equivalent of setting ip in docker-compose files? I tried ipv4_address but it did not work.

**Comment by cfelicio** [comment]
Leaving this here, as my setup is a bit tricky and I couldn't use host mode (something else is already occupying the ports). I also tried macvlan, but because I'm using hyper-v, the promiscuous mode required didn't work well (even after enabling spoofing).

In the end, I got it working with Ipvlan.  I'm also using Opnsense, and have a dedicated vlan for IOT devices, and a DHCP server, so I assigned an IP outside of the DHCP range to avoid conflicts (since Ipvlan is not able to request an ip from the DHCP server). The two code snippets from compose, in case it helps someone:

```
        networks:
          ipvlan:
            ipv4_address: 192.168.40.250    


networks:
  ipvlan:
    driver: ipvlan
    driver_opts:
      parent: eth0  # Use the actual interface name
      ipvlan_mode: l2  # Layer 2 mode
    ipam:
      driver: default
      config:
        - subnet: 192.168.40.0/24
          gateway: 192.168.40.1
```

**Reply by maxfield-allison** [reply]
I have a detailed write up with swarm and standalone docker engine along with some network settings for those using vlans here https://github.com/mrlt8/docker-wyze-bridge/discussions/1471#discussion-8189896

you may benefit from using a udp broadcast relay as detailed in one of the comments
